contrib/buildrefactor/src/python/pants/contrib/buildrefactor/peek.py

```diff
@@ -0,0 +1,44 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+#./pants peek --path=src/scala/org/pantsbuild/zinc/analysis --target-name=zinc-analysis
+#./pants peek --path=src/scala/org/pantsbuild/zinc/extractor --target-name=zinc-extractor --line_number=True
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.task.task import Task
+from pants.contrib.buildrefactor.buildozer import Buildozer
+
+
+class Peek(Task):
```

Comment:
Could you make this a subclass of [`ConsoleTask`](https://github.com/pantsbuild/pants/blob/master/src/python/pants/task/console_task.py)? I think the UX would be better.

A good example of ConsoleTask usage is [`ListRoots`](https://github.com/pantsbuild/pants/blob/d30cca1e0ecb9cc0e1b7e2cd0ff6e7e077e62a52/src/python/pants/core_tasks/roots.py#L11)

Reply:
Yes, I thought the same and tried to implement it as a ConsoleTask but have an error. Here's the current working branch with changes to `peek.py` and `test_peek_integration.py`: https://github.com/denalimarsh/pants/tree/peek-console-task

The `ConsoleTask` version of peek seems to work initially, successfully printing the target's name and line number - but directly afterwards it fails with error message:
`for value in self.console_output(targets):
Exception message: 'NoneType' object is not iterable`

Reply:
@denalimarsh consoletask's output should be in the return (or yield in some cases). https://github.com/pantsbuild/pants/blob/d30cca1e0ecb9cc0e1b7e2cd0ff6e7e077e62a52/src/python/pants/core_tasks/roots.py#L17, so don't print anything in `def console_output`, rather, make it a return value.

Reply:
Thanks for the fix! Implemented on branch peek-console-task.

Reply:
Hm. In the branch, it now returns an empty string. I think if you were to change it so that it doesn't return, but instead `yield`s where it prints, it'll work.

ie
```python
   print('\'{}\' found in BUILD file.\n'.format(root_name))
```
becomes
```python
   yield '\'{}\' found in BUILD file.\n'.format(root_name)
```

If you add the yields, the return will be unnecessary because the method will act like a generator. Does that make sense?